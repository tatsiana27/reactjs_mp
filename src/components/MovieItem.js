import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { parseReleaseDate } from '../helpers/parse-release-date';

import '../styles/movie-item.scss';
export function MovieItem(props) {
    return (

        <div className="movie-item">
            <img className="movie-item_image" src={props.image}/>
            <div className="movie-item_info">
                <Link to={`/movie-info/${props.id}`} className="movie-item_title"><div>{props.title}</div></Link>
                <div className="movie-item_release-date">{parseReleaseDate(props.releaseDate)}</div>
            </div>
            <div className="movie-item_genre">{props.genre}</div>
        </div>
    )
}
