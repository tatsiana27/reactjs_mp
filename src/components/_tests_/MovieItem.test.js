import React from 'react';
import { MovieItem } from '../MovieItem';
import { shallow } from 'enzyme';

describe('MovieItem ', () => {
    test('should render MovieItem Component', () => {
        const wrapper = shallow(
            <MovieItem/>

        );

        expect(wrapper).toMatchSnapshot();
    });
});