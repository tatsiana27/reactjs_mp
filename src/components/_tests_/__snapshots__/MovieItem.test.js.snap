// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem  should render MovieItem Component 1`] = `
<div
  className="movie-item"
>
  <img
    className="movie-item_image"
  />
  <div
    className="movie-item_info"
  >
    <div
      className="movie-item_title"
    />
    <div
      className="movie-item_release-date"
    />
  </div>
  <div
    className="movie-item_genre"
  />
</div>
`;
